### PR TITLE
Access username regex fixup

### LINKED ISSUE #714: postgresql_access raises `NoMethodError: undefined method `named_captures' for nil:NilClass`
Hi !

I noticed that when adding entries in pg_hba.conf using the `postgresql_access` resources, if there are entries where username contains a `.` then an exception is raised because parsing fails

Stacktrace is : 
```
    NoMethodError
    -------------
    undefined method `named_captures' for nil:NilClass

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/postgresql/libraries/access.rb:155:in `block in split_entries'
    /var/chef/cache/cookbooks/postgresql/libraries/access.rb:155:in `map!'
    /var/chef/cache/cookbooks/postgresql/libraries/access.rb:155:in `split_entries'
    /var/chef/cache/cookbooks/postgresql/libraries/access.rb:89:in `read!'
    /var/chef/cache/cookbooks/postgresql/libraries/access.rb:139:in `read'
    /var/chef/cache/cookbooks/postgresql/resources/access.rb:64:in `block in class_from_file'
```

I've been testing the regex at  https://github.com/sous-chefs/postgresql/blob/main/libraries/access.rb#L73 against my config file using rubular and it's not matching

Thanks for your help !

### PR DESCRIPTION
**Pull request by bmhughes**
# Description

- Fix regex not matching usernames containing characters other than word characters

## Issues Resolved

- Fixes #714

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ This file is used to list changes made in the last 3 major versions of the postg
 
 ## 11.1.1 - *2022-12-22*
 
+- Fix regex not matching usernames containing characters other than word characters
```
Comment (onlyhavecans) [inline CHANGELOG.md:9]: Should this be under unreleased, not the prior version?

Comment (ramereth) [reply CHANGELOG.md:9]: Yup, @bmhughes please fix and then I'll approve!

**Review by ramereth — CHANGES_REQUESTED** [review]
Please fix changelog

File: CHANGELOG.md (hunk 1 above)
Comment (Xorima) [reply CHANGELOG.md:9]: One day I'll update the system to use the PR body and not manual change log entries ... one day ... (I've been saying this to myself for 2+ years now)

**Comment by bmhughes** [comment]
Should be good now

**Review by onlyhavecans — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [11.1.2](https://github.com/sous-chefs/postgresql/releases/tag/11.1.2)

**Pull request merged**